include/fixed_containers/reflection.hpp

```diff
@@ -19,120 +20,216 @@ namespace fixed_containers::reflection_detail
 template <typename T>
 concept Reflectable = std::is_class_v<T> && ConstexprDefaultConstructible<T>;
 
-class DepthTracker
+enum class LayerType
 {
-    int depth_{-1};
+    ENCLOSING_FIELD,
+    BASE_CLASS,
+};
+
+struct FieldAsString
+{
+    std::string_view type_name;
+    std::string_view name;
+};
+
+inline constexpr std::string_view NULL_FIELD_TYPE_NAME{};
+
+template <std::size_t MAXIMUM_LAYERS = 32>
```

Comment:
Should we make it so that this template parameter can be provided by the client? For example, perhaps we can make `RecursionType` a variant that supports passing in recursion depth if the client wants recursive reflection info.

Reply:
Yeah, this was intended to be exposed. Let me consider a nice way to do this, as this type is not ,meant to be used by the user as it stands.